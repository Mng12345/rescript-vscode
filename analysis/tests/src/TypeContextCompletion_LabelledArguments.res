type someVariant = One | Two | Three | Four | Five(int) | Six(option<string>)

let someValue = Two

let someVariantToString = (
  ~someVariant,
  ~anotherThing: TypeDefinition.variant,
  ~thirdThing: option<int>,
) => {
  ignore(anotherThing)
  ignore(thirdThing)
  switch someVariant {
  | One => "One"
  | Two => "Two"
  | Three => "Three"
  | Four => "Four"
  | _ => "-"
  }
}

// let x = someVariantToString(~someVaria
//                                       ^com

// let x = someVariantToString(~someVariant=
//                                          ^com

// let x = someVariantToString(~someVariant=
//                                           ^com

// let x = someVariantToString(~someVariant=T
//                                           ^com

// let x = someVariantToString(~someVariant=t
//                                           ^com

// let x = someVariantToString(~anotherThing=
//                                           ^com
