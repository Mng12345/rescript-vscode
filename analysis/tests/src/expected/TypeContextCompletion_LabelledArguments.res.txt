Complete src/TypeContextCompletion_LabelledArguments.res 16:41
posCursor:[16:41] posNoWhite:[16:40] Found expr:[16:11->16:41]
Pexp_apply ...[16:11->16:30] (~someVaria16:32->16:41=...[16:32->16:41])
Completable: CnamedArg(Value[someVariantToString], someVaria, [someVaria])
Found type for function (
  ~someVariant: someVariant,
  ~anotherThing: string,
  ~thirdThing: option<int>,
) => string
[{
    "label": "someVariant",
    "kind": 4,
    "tags": [],
    "detail": "someVariant",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 19:44
posCursor:[19:44] posNoWhite:[19:43] Found expr:[19:11->19:44]
Pexp_apply ...[19:11->19:30] (~someVariant19:32->19:43=...__ghost__[0:-1->0:-1])
found typed context 
Completable: CtypedContext(Value[someVariantToString], NamedArg(someVariant))
Found variant type for NamedArg typed context someVariant
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }, {
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three",
    "kind": 4,
    "tags": [],
    "detail": "Three\n\n",
    "documentation": null
  }, {
    "label": "Four",
    "kind": 4,
    "tags": [],
    "detail": "Four\n\n",
    "documentation": null
  }, {
    "label": "Five(_)",
    "kind": 4,
    "tags": [],
    "detail": "Five(int)\n\n",
    "documentation": null
  }, {
    "label": "Six(_)",
    "kind": 4,
    "tags": [],
    "detail": "Six(option<string>)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 22:45
posCursor:[22:45] posNoWhite:[22:43] Found expr:[22:11->22:44]
Pexp_apply ...[22:11->22:30] (~someVariant22:32->22:43=...__ghost__[0:-1->0:-1])
found typed context 
Completable: CtypedContext(Value[someVariantToString], NamedArg(someVariant))
Found variant type for NamedArg typed context someVariant
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }, {
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three",
    "kind": 4,
    "tags": [],
    "detail": "Three\n\n",
    "documentation": null
  }, {
    "label": "Four",
    "kind": 4,
    "tags": [],
    "detail": "Four\n\n",
    "documentation": null
  }, {
    "label": "Five(_)",
    "kind": 4,
    "tags": [],
    "detail": "Five(int)\n\n",
    "documentation": null
  }, {
    "label": "Six(_)",
    "kind": 4,
    "tags": [],
    "detail": "Six(option<string>)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 25:45
posCursor:[25:45] posNoWhite:[25:44] Found expr:[25:11->25:45]
Pexp_apply ...[25:11->25:30] (~someVariant25:32->25:43=...[25:44->25:45])
found typed context 
Completable: CtypedContext(Value[someVariantToString], NamedArg(someVariant))
Found variant type for NamedArg typed context someVariant
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }, {
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three",
    "kind": 4,
    "tags": [],
    "detail": "Three\n\n",
    "documentation": null
  }, {
    "label": "Four",
    "kind": 4,
    "tags": [],
    "detail": "Four\n\n",
    "documentation": null
  }, {
    "label": "Five(_)",
    "kind": 4,
    "tags": [],
    "detail": "Five(int)\n\n",
    "documentation": null
  }, {
    "label": "Six(_)",
    "kind": 4,
    "tags": [],
    "detail": "Six(option<string>)\n\n",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 28:45
posCursor:[28:45] posNoWhite:[28:44] Found expr:[28:11->28:45]
Pexp_apply ...[28:11->28:30] (~someVariant28:32->28:43=...[28:44->28:45])
found typed context 
Completable: CtypedContext(Value[someVariantToString], NamedArg(someVariant))
Found variant type for NamedArg typed context someVariant
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }, {
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three",
    "kind": 4,
    "tags": [],
    "detail": "Three\n\n",
    "documentation": null
  }, {
    "label": "Four",
    "kind": 4,
    "tags": [],
    "detail": "Four\n\n",
    "documentation": null
  }, {
    "label": "Five(_)",
    "kind": 4,
    "tags": [],
    "detail": "Five(int)\n\n",
    "documentation": null
  }, {
    "label": "Six(_)",
    "kind": 4,
    "tags": [],
    "detail": "Six(option<string>)\n\n",
    "documentation": null
  }]

